project/assets/scripts/controller/details/details.js

```diff
@@ -1,49 +1,52 @@
-const DetailsEvents = function (options = {}) {
-    this.init = function (redirectToFunction) {
-        setNavEvent('new-evaluation', addEvent.bind(null, 'newEvaluation', redirectToFunction));
-        setNavEvent('evaluations', addEvent.bind(null, 'evaluations', redirectToFunction));
-        setNavEvent('logout', addEvent.bind(null, 'login', redirectToFunction));
-    };
-    this.destroy = function (redirectToFunction) {
-        removeEvents('new-evaluation', addEvent.bind(null, 'newEvaluation', redirectToFunction));
-        removeEvents('evaluations', addEvent.bind(null, 'newEvaluation', redirectToFunction));
-        removeEvents('logout', addEvent.bind(null, 'login', redirectToFunction));
-    };
-    this.populate = function (currentName) {
-        const currentCandidateData = getCurrentCandidateData(currentName);
-        console.log(currentName);
-        document.getElementById('candidate').value = currentCandidateData[0].inputData.candidate;
-        document.getElementById('interviewer').value = currentCandidateData[0].inputData.interviewer;
-        document.getElementById('date').value = currentCandidateData[0].inputData.date;
+(function () {
+    const DetailsEvents = function (options = {}) {
+        this.init = function (redirectToFunction) {
+            setNavEvent('new-evaluation', addEvent.bind(null, 'newEvaluation', redirectToFunction));
+            setNavEvent('evaluations', addEvent.bind(null, 'evaluations', redirectToFunction));
+            setNavEvent('logout', addEvent.bind(null, 'login', redirectToFunction));
+        };
+        this.destroy = function (redirectToFunction) {
+            removeEvents('new-evaluation', addEvent.bind(null, 'newEvaluation', redirectToFunction));
+            removeEvents('evaluations', addEvent.bind(null, 'newEvaluation', redirectToFunction));
+            removeEvents('logout', addEvent.bind(null, 'login', redirectToFunction));
+        };
+        this.populate = function (currentId) {
+            const currentCandidateData = getCurrentCandidateData(currentId);
+            document.getElementById('candidate').value = currentCandidateData.inputData.candidate;
```

Comment:
Now that I’m thinking about this a bit more, I wonder if it would make more sense to have this population with data inside the rendering part. Just a suggestion.

Reply:
I think it would